src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/AzureOpenAIEmbeddingClient.java

```diff
@@ -44,12 +37,15 @@ public AzureOpenAIEmbeddingClient(
       String modelName,
       int dimension,
       Map<String, Object> vectorizeServiceParameters) {
-    this.requestProperties = requestProperties;
-    this.modelName = modelName;
     // One special case: legacy "ada-002" model does not accept "dimension" parameter
-    this.dimension = EmbeddingUtil.acceptsOpenAIDimensions(modelName) ? dimension : 0;
-    String actualUrl = EmbeddingUtil.replaceParameters(baseUrl, vectorizeServiceParameters);
+    super(
```

Comment:
As per earlier comment, should probably pass `providerId` constant via constructor as well, for the abstract super class to use.